### PR TITLE
Upgrade pip dependencies

### PR DESCRIPTION
**Pull request by judgeaxl**
Several dependencies have breaking changes and major version bumps, so this change should be tested properly before merging, and changelogs for the affected modules reviewed

### CODE REVIEW

**Comment by judgeaxl** [comment]
From what I can tell there's nothing in the changes that should break anything. There's an AWS region inheritance in aws-assume-role-lib that has been changed which cannot be tested without deploying, but it looks safe too.

File: Pipfile.lock [hunk 1]
```diff
@@ -153,67 +153,73 @@
...
+                "sha256:c10c797ac89c746e488d2ee92bd4abd593615694ee17b2500578b63cad6b93a8",
+                "sha256:ced40344e811d6abba00295ced98c01aecf0c2de39481792d87af4fa58b7b4d6",
+                "sha256:d57e0cdc1b44b6cdf8af1d01807db06886f10177469312fbde8f44ccbb284bc9",
+                "sha256:d99915d6ab265c22873f1b4d6ea5ef462ef797b4140be4c9d8b179915e0985c6",
+                "sha256:eb80e8a1f91e4b7ef8b33041591e6d89b2b8e122d787e87eeb2b08da71bb16ad",
+                "sha256:ebeddd119f526bcf323a89f853afb12e225902a24d29b55fe18dd6fcb2838a76"
             ],
             "markers": "python_version >= '3.6'",
-            "version": "==3.4.7"
+            "version": "==35.0.0"
```
Comment (einarth) [inline Pipfile.lock:178]: wonder if this is a type-o ;)  looks like it should've been 3.5.0

Comment (judgeaxl) [reply Pipfile.lock:178]: They changed the numbering format

**Comment by judgeaxl** [comment]
Yeah, this can go on develop, no need to squeeze in into 0.5

**Pull request merged**